src/state_space/mod.rs

```diff
@@ -142,26 +160,52 @@ where
         StateSpaceBuilder { filters, ..self }
     }
 
-    pub async fn sync(self) -> Result<StateSpaceManager<T, N, P>, AMMError> {
-        let chain_tip = BlockId::from(self.provider.get_block_number().await?);
-        let factories = self.factories.clone();
+    pub fn with_checkpoint(self, path: impl AsRef<Path>) -> StateSpaceBuilder<T, N, P> {
+        StateSpaceBuilder {
+            checkpoint_path: Some(path.as_ref().canonicalize().unwrap()),
+            ..self
+        }
+    }
+
+    pub fn load_checkpoint(mut self) -> StateSpaceBuilder<T, N, P> {
+        if let Some(path) = self.checkpoint_path.as_ref() {
+            let state_space: StateSpace = serde_json::from_reader(File::open(path).expect(
+                "Failed to open checkpoint file. Ensure the path is correct and you have read permissions.",
+            )).expect("Failed to deserialize checkpoint file");
+            self.latest_block = state_space.latest_block.load(Ordering::Relaxed);
+            self.amms.extend(state_space.state.into_values());
+        };
+        self
+    }
+
+    pub async fn sync(self) -> Result<StateSpaceManager<T, N, P>, StateSpaceError> {
+        let this = self.load_checkpoint();
```

Comment:
Lets update to use `mut self` as well as update to `fn load_checkpoint(&mut self)` and we can use `self` here instead of 
binding to `this`.